client/src/components/PaymentMethods/PaymentMethods.tsx

```diff
@@ -0,0 +1,294 @@
+        cardName = key;
+      }
+    });
+    const expiresProperty = expiresAt.split('-');
+    let expires: string;
+    if (expiresProperty.length === 2) {
+      // To get the last two values of year in Chrome case
+      expires = `${expiresProperty[1]}/${expiresProperty[0].substring(
+        expiresProperty[0].length - 2,
+        expiresProperty[0].length,
+      )}`;
+    } else {
+      expires = expiresAt;
+    }
+    setCards((prevState) => [...prevState, { expires, cardNumber, cardName, cvv }]);
+    setSubmitting(false);
+    setIsModelOpen(false);
+  };
+
+  const validateForm = (values: FormValues) => {
+    const errors: { cardNumber?: string; expires?: string; cvv?: string } = {};
+    if (values.cardNumber) {
+      const result = validateCard(values.cardNumber);
+      if (!result) {
+        errors.cardNumber = 'Please, enter correct card number.';
+      } else {
+        setIsButtonDisabled(false);
+      }
+    }
+    if (values.cvv) {
+      const result = validateCVV(values.cardNumber, values.cvv);
+      if (!result) {
+        errors.cvv = 'Please, enter correct CVV';
+      } else {
+        setIsButtonDisabled(false);
+      }
+    }
+    return errors;
+  };
+
+  const formatCardNumber = (valueArg: string): string => {
+    // remove all non digit characters
+    const value = valueArg.replace(/\D/g, '');
+    let formattedValue;
+    // american express, 15 digits
+    if (/^3[47]\d{0,13}$/.test(value)) {
+      formattedValue = value.replace(/(\d{4})/, '$1 ').replace(/(\d{4}) (\d{6})/, '$1 $2 ');
+      setMaxLength((prevState) => ({ ...prevState, cardNumber: 17 }));
+    } else if (/^3(?:0[0-5]|[68]\d)\d{0,11}$/.test(value)) {
+      // diner's club, 14 digits
+      formattedValue = value.replace(/(\d{4})/, '$1 ').replace(/(\d{4}) (\d{6})/, '$1 $2 ');
+      setMaxLength((prevState) => ({ ...prevState, cardNumber: 16 }));
+    } else if (/^\d{0,16}$/.test(value)) {
+      // regular cc number, 16 digits
+      formattedValue = value.match(/.{1,4}/g)?.join(' ');
+      setMaxLength((prevState) => ({ ...prevState, cardNumber: 19 }));
+    }
+    return formattedValue || '';
+  };
+
+  return (
+    <>
+      <CardHeader title="Payment Methods" component="h1" className={classes.cardHeader} />
+      <CardContent className={classes.cardContent}>
+        {cards.length ? (
+          <>
+            <Typography color="textSecondary" className={classes.marginLeft} align={isLessthanSm ? 'center' : 'left'}>
+              Saved Payment Profiles:
+            </Typography>
+            <Grid container spacing={2} className={classes.cardsContainer}>
+              {cards.map((card, idx) => (
+                <Grid item xs={12} md={6} key={idx}>
+                  <Card elevation={1}>
+                    <CardHeader
+                      avatar={
+                        <PaymentIcon id={card.cardName} style={{ margin: 10, width: 100 }} className="payment-icon" />

```

Comment:
Try to avoid inline-styling whenever possible

Reply:
Will do that.